### PR TITLE
[UI-130] Create multi-checkbox component

### PR DESCRIPTION
**Pull request by teamchong**
image

Closes [UI-130](https://eventstore.aha.io/develop/features/UI-130)

### CODE REVIEW

File: packages/fields/src/components/es-multi-checkbox/es-multi-checkbox.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+    componentWillLoad() {
+        this.internalValue = { ...this.value };
+    }
+
+    render() {
+        return (
+            <div class={{ field: true, invalid: !!this.invalid }}>
+                {this.options.map(option => (
+                    <label>
```
Comment (George-Payne) [inline packages/fields/src/components/es-multi-checkbox/es-multi-checkbox.tsx:41]: Maybe needs a key

**Review by George-Payne — CHANGES_REQUESTED** [review]
In testing, it seems like it's completely bypassed by keyboard navigation. 

https://github.com/user-attachments/assets/1351e7d0-4d37-43e4-bdd8-a5046c0adf5a

I'd expect it to work like a [fieldset](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset) tab to focus, and arrows to move between the items.

File: tools/stencil-markdown-plugin/plugin.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
+"use strict";
```
Comment (George-Payne) [inline tools/stencil-markdown-plugin/plugin.js:1]: Was this file intended to be added?

File: packages/fields/src/components/es-multi-checkbox/es-multi-checkbox.tsx [hunk 3]
```diff
@@ -37,8 +37,8 @@ export class EsMultiCheckbox {
     render() {
         return (
             <div class={{ field: true, invalid: !!this.invalid }}>
-                {this.options.map(option => (
-                    <label>
+                {this.options.map((option, i) => (
+                    <label key={option.value} tabIndex={i}>
```
Comment (George-Payne) [inline packages/fields/src/components/es-multi-checkbox/es-multi-checkbox.tsx]: `tabindex` should be `0`, also needs to be `undefined` if the field is disabled.

File: packages/fields/src/components/es-multi-checkbox/es-multi-checkbox.tsx [hunk 4]
```diff
@@ -1,70 +1,118 @@
...
 @Component({
     tag: 'es-multi-checkbox',
     styleUrl: 'es-multi-checkbox.css',
     shadow: true,
 })
 export class EsMultiCheckbox {
     /** Emitted when the value of the field is changed. */
-    @Event({ bubbles: true }) fieldchange!: EventEmitter<FieldChange<Record<string, boolean>>>;
+    @Event({ bubbles: true }) fieldchange!: EventEmitter<
+        FieldChange<string | null>
```
Comment (George-Payne) [inline packages/fields/src/components/es-multi-checkbox/es-multi-checkbox.tsx:19]: Why did this get changed to `string / null`? I don't see anything about it in the commit message. I think the previous interface seemed nicer to use, did something not work with it?

**Review by George-Payne — CHANGES_REQUESTED** [review]
Some other issues:
[ ] Demo no-longer works, as the components interface was changed, but the demo wasn't updated
[ ] Toggling with keyboard doesn't update the form
[ ] Match focus styles to https://github.com/EventStore/Design-System/pull/384

File: packages/fields/src/components/es-multi-checkbox/es-multi-checkbox.tsx (hunk 4 above)
Comment (George-Payne) [reply packages/fields/src/components/es-multi-checkbox/es-multi-checkbox.tsx:19]: Ah, yeah, that's for nested forms, maybe there's a better way to do it, but best not to mess with it now.

My only concern with that interface is that it will break if we pass an option with a new-line in it. It's probably an unfounded concern, but might bite us later.

How about `Set<string>`? if it's in the set, it's checked, if not, it's not. 

The change function becomes a bit simpler also:

```
        const { value, checked } = e.target as HTMLInputElement;

        if (!value) return;

        let newValue = new Set(this.value);

        if (checked) {
            newValue.add(value);
        } else {
            newValue.delete(value);
        }

        this.fieldchange.emit({
            name: this.name,
            value: newValue,
        });
 ```
 
 a `Map<string, boolean>` could also work:
 
 ```
         const { value, checked } = e.target as HTMLInputElement;

        if (!value) return;

        let newValue = new Map(this.value);

        
        newValue.set(value, checked); 


        this.fieldchange.emit({
            name: this.name,
            value: newValue,
        });
 ````
 
 But seeing if it's checked on the outside is a bit more complicated. 
 
 ```
 const checked = data.thingAsMap.get("name") ?? false;
 ```
 
 vs
 
  ```
 const checked = data.thingAsSet.has("name");
 ```

Comment (teamchong) [reply packages/fields/src/components/es-multi-checkbox/es-multi-checkbox.tsx:19]: I believe Set and Map are also object, and will have the same issue

Comment (George-Payne) [reply packages/fields/src/components/es-multi-checkbox/es-multi-checkbox.tsx:19]: eugh, you're right. This should be all we need to allow it to work:

https://github.com/EventStore/Design-System/pull/386

Comment (teamchong) [reply packages/fields/src/components/es-multi-checkbox/es-multi-checkbox.tsx]: I've also updated this because it appears to have the same logic.
image

File: tools/stencil-markdown-plugin/plugin.js (hunk 2 above)
Comment (teamchong) [reply tools/stencil-markdown-plugin/plugin.js]: Removed

**Review by George-Payne — APPROVED** [review]
